src/MetaMorpho.sol

```diff
@@ -273,18 +268,15 @@ contract MetaMorpho is ERC4626, ERC20Permit, Ownable2Step, Multicall, IMetaMorph
     /// @inheritdoc IMetaMorphoBase
     function submitCap(MarketParams memory marketParams, uint256 newSupplyCap) external onlyCuratorRole {
         Id id = marketParams.id();
+        uint256 supplyCap = config[id].cap;
         if (marketParams.loanToken != asset()) revert ErrorsLib.InconsistentAsset(id);
         if (MORPHO.lastUpdate(id) == 0) revert ErrorsLib.MarketNotCreated();
-
-        uint256 supplyCap = config[id].cap;
```

Comment:
Why moving up the SLOAD?
`supplyCap` is not used until L280

Reply:
By the same rule, id should be assigned 1 line later

Reply:
I would block this PR bnecause of this

Reply:
I'd be in favor of reverting back this change as well

Reply:
we don't usually for gas cost of reverts right?

Reply:
Yes but also why assign variables too early (wrt when they are actually used)?

Reply:
I find it ugly and useless but ok if you want I'm changing